### PR TITLE
Add generic webhook provider

### PR DESCRIPTION
**Pull request by willemm**
This provider allows a secretstore with a generic url (templated)
which will be called with a defined method, headers (templated)
and optional body (also templated)
The response can be parsed out with a jsonPath expression

Closes https://github.com/external-secrets/external-secrets/issues/548

### CODE REVIEW

**Comment by knelasevero** [comment]
LGTM :heart:

**Comment by moolen** [comment]
Hey, i was on vacation, i'll take a look asap!

**Review by moolen — APPROVED** [review]
Hey, your approach looks very good! I have two nits that you can fix now or i can create an follow-up issue and someone cleans that up later if you're short on time.

We also need a little bit of documentation for users so they know how to use the webhook provider - same as above - i can create a follow up issue for that.

File: pkg/provider/webhook/webhook.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+	})
+}
+
+func (p *Provider) NewClient(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (provider.SecretsClient, error) {
+	whClient := &WebHook{
+		kube:      kube,
+		store:     store,
+		namespace: namespace,
+		storeKind: store.GetObjectKind().GroupVersionKind().Kind,
+	}
```
[the remark this answers is not in the document]
Comment (willemm) [reply pkg/provider/webhook/webhook.go:64]: Sounds like a good idea, I wasn't aware that keeping the http.Client around would keep/reuse the TCP connection.

~~Tbh, I didn't even know you could keep the tcp connection open in http, does that get closed after a period of disuse or something similar?~~

I'll look into it tomorrow, ~~although I'll have to think about if it works in the (unlikely) event that the cacerts secret gets changed.~~

Oh wait I think I get it, the WebHook struct gets constructed each time one whole externalsecret gets reconciled, so it gets the cacerts and opens the connection once per externalsecret reconciliation, right?

**Comment by willemm** [comment]
Great!

> We also need a little bit of documentation for users so they know how to use the webhook provider - same as above - i can create a follow up issue for that.

I'll have a look tomorrow how far I can get with documentation.  Do you have a preferred provider documentation I can nick the layout from?

File: pkg/provider/webhook/webhook.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (moolen) [reply pkg/provider/webhook/webhook.go:65]: yeah, exactly. I'm refering to this part here: https://github.com/external-secrets/external-secrets/blob/main/pkg/controllers/externalsecret/externalsecret_controller.go#L391-L407 where we can reuse the http client.

**Comment by nmajin** [comment]
Thank you for adding this provider, I am using it at the moment for an external service to get an OAuth token and storing it in a secret.

One thing I cannot get by on this though is brought up here: https://github.com/external-secrets/external-secrets/issues/809 . Any idea how to avoid having to put sensitive data into the body of the POST? I saw in the example in the docs that we could reference a secret and interpolate the secrets details but this is not working or I could be doing something wrong.

**Pull request merged**